core/src/test/java/org/elasticsearch/action/update/UpdateResponseTests.java

```diff
@@ -81,29 +85,59 @@ public void testToXContent() throws IOException {
-
+        BytesReference originalBytes = toShuffledXContent(updateResponse, xContentType, ToXContent.EMPTY_PARAMS, humanReadable);
+
+        BytesReference mutated;
+        if (addRandomFields) {
+            // - The ShardInfo.Failure's exception is rendered out in a "reason" object. We shouldn't add anything random there
+            // because exception rendering and parsing are very permissive: any extra object or field would be rendered as
+            // a exception custom metadata and be parsed back as a custom header, making it impossible to compare the results
+            // in this test.
+            // - The GetResult's "_source" and "fields" just consists of key/value pairs, we shouldn't add anything random there.
+            // It is already randomized in the randomGetResult() method anyway. Also, we cannot add anything within the "get"
+            // object since this is where GetResult's metadata fields are rendered out and they would be parsed back as
+            // extra metadata fields.
+            Predicate<String> excludeFilter = path -> path.contains("reason") || path.contains("get");
+            mutated = insertRandomFields(xContentType, originalBytes, excludeFilter, random());
+        } else {
+            mutated = originalBytes;
+        }
         UpdateResponse parsedUpdateResponse;
-        try (XContentParser parser = createParser(xContentType.xContent(), updateResponseBytes)) {
+        try (XContentParser parser = createParser(xContentType.xContent(), mutated)) {
             parsedUpdateResponse = UpdateResponse.fromXContent(parser);
             assertNull(parser.nextToken());
         }
 
-        // We can't use equals() to compare the original and the parsed delete response
-        // because the random delete response can contain shard failures with exceptions,
-        // and those exceptions are not parsed back with the same types.
-        assertUpdateResponse(expectedUpdateResponse, parsedUpdateResponse);
-    }
+        IndexResponseTests.assertDocWriteResponse(expectedUpdateResponse, parsedUpdateResponse);
+        if (addRandomFields == false) {
+            assertEquals(expectedUpdateResponse.getGetResult(), parsedUpdateResponse.getGetResult());
+        }
 
-    public static void assertUpdateResponse(UpdateResponse expected, UpdateResponse actual) {
-        IndexResponseTests.assertDocWriteResponse(expected, actual);
-        assertEquals(expected.getGetResult(), actual.getGetResult());
+        // Prints out the parsed UpdateResponse object to verify that it is the same as the expected output.
+        // If random fields have been inserted, it checks that they have been filtered out and that they do
+        // not alter the final output of the parsed object.
+        BytesReference parsedBytes = toXContent(parsedUpdateResponse, xContentType, humanReadable);
```

Comment:
We don't want to shuffle here (though we could) because we just want to check that the parsed update response, when printed out as xcontent and parsed back as a map, has the exact same output as what we expect. I hoped the comment was clear enough...